Libraries/LibWeb/Layout/SVGFormattingContext.cpp

```diff
@@ -176,9 +178,33 @@ void SVGFormattingContext::run(AvailableSpace const& available_space)
     // NOTE: SVG doesn't have a "formatting context" in the spec, but this is the most
     //       obvious way to drive SVG layout in our engine at the moment.
 
+    bool is_standalone_svg_document = !this->context_box()               // the [context] box
+                                           .root()                       // ...of the [root] viewport
+                                           .document()                   // ...of the document
+                                           .page()                       // ...on the page
+                                           .top_level_browsing_context() // ...in the browser
+                                           .active_document()            // the active document (in the browser)
+                                           ->is_initial_about_blank();   // is it about:blank initially?
+    //  NOTE: Given that somehow we have already made it to an SVGFormattingContext:
+    //        The browser's active document [...] is initially about:blank can only mean that we are
+    //        laying out a fragment of another document.
+    //        Rephrase: we are not laying out a standalone SVG document.
+    //        Current browser behavior seems to guarantee initial about:blank for layout of SVG fragments.
+
     auto& svg_viewport = dynamic_cast<SVG::SVGViewport const&>(*context_box().dom_node());
     auto& svg_box_state = m_state.get_mutable(context_box());
 
+    if (is_standalone_svg_document) {
+        // Overwrite the content width/height with the styled node width/height (from <svg width height ...>)
+
+        // NOTE: If a height had not been provided by the svg element, it was set to the height of the container
+        //       (see BlockFormattingContext::layout_viewport)
+        if (svg_box_state.node().computed_values().width().is_length())
+            svg_box_state.set_content_width(svg_box_state.node().computed_values().width().length().absolute_length_to_px());
+        if (svg_box_state.node().computed_values().height().is_length())
+            svg_box_state.set_content_height(svg_box_state.node().computed_values().height().length().absolute_length_to_px());
```

Comment:
- The length might not be absolute, so you can't call `absolute_length_to_px()` but need to use one of the `to_px()` methods.
- In SVG 2, the length can also be a percentage. You don't have to handle that now, but please at least add a FIXME comment.